app/src/androidTest/java/org/mozilla/focus/activity/screenshots/SettingsScreenshots.java

```diff
@@ -99,9 +118,68 @@ public void takeScreenShotsOfSettings() throws Exception {
         Screengrab.screenshot("SearchEngine_Add_Search_Engine");
         TestHelper.pressBackKey();
+        TestHelper.pressBackKey();
 
-        /* scroll down */
+/* Tap autocomplete menu */
+        final String urlAutocompletemenu = getString(R.string.preference_subitem_autocomplete);
+        onData(withTitleText(urlAutocompletemenu))
+                .perform(click());
+        onView(withText(getString(R.string.preference_autocomplete_custom_summary)))
+                .check(matches(isDisplayed()));
+        Screengrab.screenshot("Autocomplete_Menu_Item");
+
+        /* Enable Autocomplete, and enter Custom URLs */
+        onView(withText(getString(R.string.preference_autocomplete_custom_summary)))
+                .perform(click());
+        /* Add custom URL */
+        onView(withText(getString(R.string.preference_autocomplete_subitem_customlist)))
+                .perform(click());
+        device.waitForIdle();
+        final String addCustomURLAction = getString(R.string.preference_autocomplete_action_add);
+        onView(withText(addCustomURLAction))
+                .check(matches(isDisplayed()));
+        Screengrab.screenshot("Autocomplete_Custom_URL_List");
+
+        onView(withText(addCustomURLAction))
+                .perform(click());
+        onView(withId(R.id.domainView))
+                .check(matches(isDisplayed()));
+        Screengrab.screenshot("Autocomplete_Add_Custom_URL_Dialog");
+        onView(withId(R.id.domainView))
+                .perform(typeText("screenshot.com"), closeSoftKeyboard());
+        onView(withId(R.id.save))
+                .perform(click());
+        device.waitForIdle();
+        onView(withText(addCustomURLAction))
+                .check(matches(isDisplayed()));
+
+        /* Remove menu */
+        final String removeMenu = getString(R.string.preference_autocomplete_menu_remove);
+        openActionBarOverflowOrOptionsMenu(InstrumentationRegistry.getContext());
+        device.waitForIdle();
+        onView(withText(removeMenu))
+                .check(matches(isDisplayed()));
+        Screengrab.screenshot("Autocomplete_Custom_URL_Remove_Menu_Item");
+        onView(withText(removeMenu))
+                .perform(click());
+        device.waitForIdle();
+        /* Remove dialog */
+        onView(withText(getString(R.string.preference_autocomplete_title_remove)))
+                .check(matches(isDisplayed()));
+        Screengrab.screenshot("Autocomplete_Custom_URL_Remove_Dialog");
         TestHelper.pressBackKey();
+        device.waitForIdle();
+        onView(withText(addCustomURLAction))
+                .check(matches(isDisplayed()));
+        TestHelper.pressBackKey();
+        device.waitForIdle();
+        onView(withText(urlAutocompletemenu))
+                .check(matches(isDisplayed()));
+        TestHelper.pressBackKey();
+        device.waitForIdle();
+        TestHelper.settingsHeading.waitForExists(waitingTime);
```

Comment:
Here and below you are now starting to use UIAutomator. To avoid side effects we should try to use only one of them in a single test. For tests that do not leave the app this should be Espresso if possible.

Reply:
huh, I subconciously uses uiautomator there.  will fix that.